### ISSUE TITLE
perform form.removeAll() throw Array out of bounds when have lot of Sections in Form 

### ISSUE THREAD

**Issue by hopy11** [xmartlabs/Eureka#1643]
When a lots of sections in a form (500+),perform form.removeAll() will throw array out of bounds at ：

```
extension Form: MutableCollection {

    // MARK: MutableCollectionType

    public subscript (_ position: Int) -> Section {
        get {
            //position equal kvoWrapper.section.count ,so throw out of bounds...
            return kvoWrapper.sections[position] as! Section
        }
}
```

But when there are few of sections in form,It's all right!!!
Why is that?thanks ;)

**Comment by hopy11** [comment]
And I noticed when removed every single section it have to call hasBeenRemoveSection(),and in it,it will refresh tableView.
Can this result in wasted performance???
Is there have some batch method for form.removeAll()?? It will refresh tableView once!???

**Comment by ialimz** [comment]
I have this issue too.
it causes about 400 crash on our app since last week

**Comment by ialimz** [comment]
@mats-claassen in my case, I generate a new form after call previous form's `form.removeAll()`,
and in `heightForRowAtIndexPath` (in `Core.swift`) when it wants get the row by row[indexPath.section][indexPath.row], crash happens due to `kvoWrapper.sections.count` is 0 and not updated yet

**Comment by ialimz** [comment]
@mats-claassen I overwrite `sectionsHaveBeenRemoved` in my FormViewController class with empty body to do nothing, it seems that problem fixed

**Comment by ialimz** [comment]
Nope, my tests fails on reloadRows now

**Comment by mats-claassen** [comment]
Have you tried just replacing the form of the FormViewController, without calling `removeAll`?

**Comment by rlopez1990** [comment]
I have the same problem, does anyone know the best way to solve this problem without memory leaks?

**Comment by mats-claassen** [comment]
Fixed in #1758
